### PR TITLE
fix(form): show error at the end if inputs are not valid

### PR DESCRIPTION
**Pull request by NicoSerranoP**
**Problem**

The `Form` component uses pages for forms that have multiple sections. In each section the inputs already alert the user about some conditions (e.g.: you need a valid eth address, this field needs at least 3 words, etc). 

The problem happens in the end. If the user ignored the warnings of each input in each section, then the form is not valid and cannot be submitted. When the user press the submit button nothing happens (no alert).

**Solution**

Detect if the form has any validation errors and tell the user that when he presses the submit button. That way the user will know he needs to trace back and find the input validation error.

**Extras**

1. Delete "Applications" from top navbar
2. "Create New Application" button in each round sends to the `/rounds/[roundId]/new` page

### CODE REVIEW

File: packages/interface/src/components/ui/Form.tsx
Comment (0xmad) [inline packages/interface/src/components/ui/Form.tsx:244]: handlers should not be inside jsx render. Use callbacks instead.

**Review by 0xmad — CHANGES_REQUESTED** [review]
@NicoSerranoP thanks, just one comment

File: packages/interface/src/components/ui/Form.tsx
Comment (NicoSerranoP) [reply packages/interface/src/components/ui/Form.tsx]: Done 🫡

**Comment by NicoSerranoP** [comment]
> Great! Thanks! Just thinking about some details:
> 
> 1. I think clicking the `ImageUpload` somehow trigger the `submit` function, so if there's any errors on the page, the error would be fixed at the bottom of the form, maybe check the button type of `ImageUpload`
> 2. even after I fixed the errors (e.g. fix the length of the name) the error is stilly displayed
> 3. just suggestion, thinking about changing those error reminder to other color rather than just black
> 4. another suggestion, also thinking about move the `errorMessage` to alignRight under the submit buttons
> 5. Also thinking about should we also check the restrictions before clicking `Next`?


- [ ] 1. I think it is not doing that anymore. Could you check again? 😊 
- [ ]  2. WIP
- [X]  3. should we use red in all `<ErrorMessage>` ?
- [X] 4. Done
- [ ] 5. WIP

File: packages/interface/src/components/ui/Form.tsx
Comment (0xmad) [inline packages/interface/src/components/ui/Form.tsx:243]: The dependencies should not be empty

Comment (NicoSerranoP) [reply packages/interface/src/components/ui/Form.tsx:243]: Sorry I just added the dependencies :)

File: packages/interface/src/components/ui/Form.tsx
Comment (0xmad) [inline packages/interface/src/components/ui/Form.tsx:252]: Timeout should  be cleared and timeout value should be moved to constant.

```suggestion
  );
 
  useEffect(() => {
    if (errorMessage) {
      setTimeout(() => {
        setErrorMessage(null);
      }, 7000);
    }
  }, [errorMessage]);
```

**Comment by kittybest** [comment]
> > Great! Thanks! Just thinking about some details:
> > 
> > 1. I think clicking the `ImageUpload` somehow trigger the `submit` function, so if there's any errors on the page, the error would be fixed at the bottom of the form, maybe check the button type of `ImageUpload`
> > 2. even after I fixed the errors (e.g. fix the length of the name) the error is stilly displayed
> > 3. just suggestion, thinking about changing those error reminder to other color rather than just black
> > 4. another suggestion, also thinking about move the `errorMessage` to alignRight under the submit buttons
> > 5. Also thinking about should we also check the restrictions before clicking `Next`?
> 
> * [x]  1. I think it is not doing that anymore. Could you check again? 😊
> * [x]   2. I used a timer that will delete the message after 5s (I cannot detect the changes of the previous page form without drastically changing the component instruction)
> * [x]   3. should we use red in all `<ErrorMessage>` ?
> * [x]  4. Done
> * [ ]  5. **Blocked:** I don't know how to do this in a generic mode (the steps and pages are set in each specific form component)

This 5th one is not required, just a suggestion, no need to spend much time on it.
And for the condition mentioned in 1st one, let me add a video:
https://drive.google.com/file/d/18jyLWRraw03LilH_H2KstTuCEfVG7HwF/view?usp=sharing

File: packages/interface/src/components/ui/Form.tsx
Comment (NicoSerranoP) [reply packages/interface/src/components/ui/Form.tsx:252]: Done 🚀

**Comment by NicoSerranoP** [comment]
@kittybest @0xmad @ctrlc03 I fixed two ui issues that are being addressed at #407 as well. I think the only component that would give us conflict when merging that branch is `../interface/src/features/applications/components/ApplicationsToApprove.tsx` :)

File: packages/interface/src/components/ImageUpload.tsx
Comment (0xmad) [inline packages/interface/src/components/ImageUpload.tsx:38]: Why do you need this function?

File: packages/interface/src/components/ui/Form.tsx
Comment (0xmad) [inline packages/interface/src/components/ui/Form.tsx:248]:
```suggestion
  );
  
  useEffect(() => {
```

File: packages/interface/src/components/ImageUpload.tsx
Comment (NicoSerranoP) [reply packages/interface/src/components/ImageUpload.tsx:38]: To prevent the form to be submitted (and show the validation errors) when clicking on the image

Comment (kittybest) [reply packages/interface/src/components/ImageUpload.tsx:38]: to solve this problem: https://github.com/privacy-scaling-explorations/maci-platform/pull/398#issuecomment-2419999794

**Review by kittybest — APPROVED** [review]
Currently lgtm, thanks a lot! plz rebase and fix the conflicts

**Comment by NicoSerranoP** [comment]
I think we need @0xmad 's approval for the code changes request 
image

**Comment by kittybest** [comment]
@NicoSerranoP There's still prettier and commitlint check not passed.
So you need to:
1. run `pnpm lint:ts` in the package first
2. squash the commits, that there wouldn't be any `Merge branch 'main' into fix/validation-errors-in-form` anymore, which is causing the commitlint error.
3. to merge main branch in the future, please use `rebase` instead of `merge`

**Comment by NicoSerranoP** [comment]
@kittybest I think I finally have it ready to be merged 🥳.

@0xmad could you help me approving the "change requested" to activate the merging option?

**Comment by 0xmad** [comment]
@NicoSerranoP you still need to squash all the commits into one. After that it can be merged.

**Review by 0xmad — APPROVED** [review]
@NicoSerranoP thanks!

**Pull request merged**
